### PR TITLE
docs: changelog 5.22.2

### PR DESCRIPTION
**Pull request by MadCcc**
- 💄 Tweak Button style code. [#51646](https://github.com/ant-design/ant-design/pull/51646) [@afc163](https://github.com/afc163)
- 🐞 Fix Input.OTP focus from advancing when previous input is empty. [#51664](https://github.com/ant-design/ant-design/pull/51664) [@thecodesalim](https://github.com/thecodesalim)
- 🐞 Adjust Modal function call not to scroll the confirm button when it get auto focused. [#51647](https://github.com/ant-design/ant-design/pull/51647) [@zombieJ](https://github.com/zombieJ)
- 🐞 Fix Form `rules` with same error content will case React render warning. [#51636](https://github.com/ant-design/ant-design/pull/51636) [@zombieJ](https://github.com/zombieJ)
- 🐞 Refactor Button `focus` logic trigger with `useEffect` to resolve some async load case not get `autoFocus`. [#51624](https://github.com/ant-design/ant-design/pull/51624) [@zombieJ](https://github.com/zombieJ)
- 🌐 Added support for Russian translation. [#51619](https://github.com/ant-design/ant-design/pull/51619) [@avvakumovid](https://github.com/avvakumovid)

---

- 💄 优化 Button 的样式。[#51646](https://github.com/ant-design/ant-design/pull/51646) [@afc163](https://github.com/afc163)
- 🐞 修复 Input.OTP 组件在有非法输入时仍会切换到下一个输入框的问题。[#51664](https://github.com/ant-design/ant-design/pull/51664) [@thecodesalim](https://github.com/thecodesalim)
- 🐞 调整 Modal 确认函数，使其在弹出后聚焦确认按钮时，不要滚动窗体。[#51647](https://github.com/ant-design/ant-design/pull/51647) [@zombieJ](https://github.com/zombieJ)
- 🐞 修复 Form `rules` 生成多条相同错误时，会报 React 渲染错误的问题。[#51636](https://github.com/ant-design/ant-design/pull/51636) [@zombieJ](https://github.com/zombieJ)
- 🐞 调整 Button 使用 `useEffect` 来触发 `autoFocus` 逻辑以解决一些异步渲染场景下，Button 无法自动聚焦的问题。[#51624](https://github.com/ant-design/ant-design/pull/51624) [@zombieJ](https://github.com/zombieJ)
- 🌐 添加了俄语翻译支持。[#51619](https://github.com/ant-design/ant-design/pull/51619) [@avvakumovid](https://github.com/avvakumovid)

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### CODE REVIEW

File: CHANGELOG.zh-CN.md [hunk 1]
```diff
@@ -15,6 +15,17 @@ tag: vVERSION
 
 ---
 
+## 5.22.2
+
+`2024-11-18`
+
+- 💄 优化 Button 的样式。[#51646](https://github.com/ant-design/ant-design/pull/51646) [@afc163](https://github.com/afc163)
```
Comment (afc163) [inline CHANGELOG.zh-CN.md:22]: 不用写，代码风格重构，对用户无影响。

File: CHANGELOG.zh-CN.md [hunk 2]
```diff
@@ -15,6 +15,17 @@ tag: vVERSION
...
+- 🐞 修复 Input.OTP 组件在有非法输入时仍会切换到下一个输入框的问题。[#51664](https://github.com/ant-design/ant-design/pull/51664) [@thecodesalim](https://github.com/thecodesalim)
+- 🐞 调整 Modal 确认函数，使其在弹出后聚焦确认按钮时，不要滚动窗体。[#51647](https://github.com/ant-design/ant-design/pull/51647) [@zombieJ](https://github.com/zombieJ)
+- 🐞 修复 Form `rules` 生成多条相同错误时，会报 React 渲染错误的问题。[#51636](https://github.com/ant-design/ant-design/pull/51636) [@zombieJ](https://github.com/zombieJ)
```
Comment (afc163) [inline CHANGELOG.zh-CN.md:25]:
```suggestion
- 🐞 修复 Form `rules` 生成多条相同错误时会报 React 渲染错误的问题。[#51636](https://github.com/ant-design/ant-design/pull/51636) [@zombieJ](https://github.com/zombieJ)
```

File: CHANGELOG.zh-CN.md [hunk 3]
```diff
@@ -15,6 +15,17 @@ tag: vVERSION
...
+- 🐞 调整 Button 使用 `useEffect` 来触发 `autoFocus` 逻辑以解决一些异步渲染场景下，Button 无法自动聚焦的问题。[#51624](https://github.com/ant-design/ant-design/pull/51624) [@zombieJ](https://github.com/zombieJ)
+- 🌐 添加了俄语翻译支持。[#51619](https://github.com/ant-design/ant-design/pull/51619) [@avvakumovid](https://github.com/avvakumovid)
```
Comment (afc163) [inline CHANGELOG.zh-CN.md:27]:
```suggestion
- 🇷🇺 添加了俄语翻译支持。[#51619](https://github.com/ant-design/ant-design/pull/51619) [@avvakumovid](https://github.com/avvakumovid)
```

File: CHANGELOG.zh-CN.md [hunk 4]
```diff
@@ -15,6 +15,17 @@ tag: vVERSION
...
 
+## 5.22.2
+
+`2024-11-18`
+
+- 💄 优化 Button 的样式。[#51646](https://github.com/ant-design/ant-design/pull/51646) [@afc163](https://github.com/afc163)
+- 🐞 修复 Input.OTP 组件在有非法输入时仍会切换到下一个输入框的问题。[#51664](https://github.com/ant-design/ant-design/pull/51664) [@thecodesalim](https://github.com/thecodesalim)
+- 🐞 调整 Modal 确认函数，使其在弹出后聚焦确认按钮时，不要滚动窗体。[#51647](https://github.com/ant-design/ant-design/pull/51647) [@zombieJ](https://github.com/zombieJ)
+- 🐞 修复 Form `rules` 生成多条相同错误时，会报 React 渲染错误的问题。[#51636](https://github.com/ant-design/ant-design/pull/51636) [@zombieJ](https://github.com/zombieJ)
+- 🐞 调整 Button 使用 `useEffect` 来触发 `autoFocus` 逻辑以解决一些异步渲染场景下，Button 无法自动聚焦的问题。[#51624](https://github.com/ant-design/ant-design/pull/51624) [@zombieJ](https://github.com/zombieJ)
```
Comment (afc163) [inline CHANGELOG.zh-CN.md:26]:
```suggestion
- 🐞 调整 Button 使用 `useEffect` 来触发 `autoFocus` 逻辑，以解决一些异步渲染场景下 Button 无法自动聚焦的问题。[#51624](https://github.com/ant-design/ant-design/pull/51624) [@zombieJ](https://github.com/zombieJ)
```

File: CHANGELOG.en-US.md [hunk 5]
```diff
@@ -15,6 +15,17 @@ tag: vVERSION
...
 ---
 
+## 5.22.2
+
+`2024-11-18`
+
+- 💄 Tweak Button style code. [#51646](https://github.com/ant-design/ant-design/pull/51646) [@afc163](https://github.com/afc163)
+- 🐞 Fix Input.OTP focus from advancing when previous input is empty. [#51664](https://github.com/ant-design/ant-design/pull/51664) [@thecodesalim](https://github.com/thecodesalim)
+- 🐞 Adjust Modal function call not to scroll the confirm button when it get auto focused. [#51647](https://github.com/ant-design/ant-design/pull/51647) [@zombieJ](https://github.com/zombieJ)
+- 🐞 Fix Form `rules` with same error content will case React render warning. [#51636](https://github.com/ant-design/ant-design/pull/51636) [@zombieJ](https://github.com/zombieJ)
```
[the remark this answers is not in the document]
Comment (afc163) [reply CHANGELOG.en-US.md:25]: +1

Comment (MadCcc) [inline CHANGELOG.en-US.md:25]:
```suggestion
- 🐞 Fix Form `rules` with same error content will cause React render warning. [#51636](https://github.com/ant-design/ant-design/pull/51636) [@zombieJ](https://github.com/zombieJ)
```

File: CHANGELOG.zh-CN.md (hunk 1 above)
Comment (MadCcc) [inline CHANGELOG.zh-CN.md:22]:
```suggestion
```

File: CHANGELOG.en-US.md [hunk 6]
```diff
@@ -15,6 +15,17 @@ tag: vVERSION
 
 ---
 
+## 5.22.2
+
+`2024-11-18`
+
+- 💄 Tweak Button style code. [#51646](https://github.com/ant-design/ant-design/pull/51646) [@afc163](https://github.com/afc163)
```
Comment (MadCcc) [inline CHANGELOG.en-US.md:22]:
```suggestion
```

File: CHANGELOG.en-US.md [hunk 7]
```diff
@@ -15,6 +15,16 @@ tag: vVERSION
...
 
+## 5.22.2
+
+`2024-11-18`
+
+- 🐞 Fix Input.OTP focus from advancing when previous input is empty. [#51664](https://github.com/ant-design/ant-design/pull/51664) [@thecodesalim](https://github.com/thecodesalim)
+- 🐞 Adjust Modal function call not to scroll the confirm button when it get auto focused. [#51647](https://github.com/ant-design/ant-design/pull/51647) [@zombieJ](https://github.com/zombieJ)
+- 🐞 Fix Form `rules` with same error content will cause React render warning. [#51636](https://github.com/ant-design/ant-design/pull/51636) [@zombieJ](https://github.com/zombieJ)
+- 🐞 Refactor Button `focus` logic trigger with `useEffect` to resolve some async load case not get `autoFocus`. [#51624](https://github.com/ant-design/ant-design/pull/51624) [@zombieJ](https://github.com/zombieJ)
+- 🌐 Added support for Russian translation. [#51619](https://github.com/ant-design/ant-design/pull/51619) [@avvakumovid](https://github.com/avvakumovid)
```
Comment (thinkasany) [inline CHANGELOG.en-US.md:26]:
```suggestion
- 🇷🇺 Added support for Russian translation. [#51619](https://github.com/ant-design/ant-design/pull/51619) [@avvakumovid](https://github.com/avvakumovid)
```

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/pull/51652

**Comment by li-jia-nan** [comment]
- #51661

**Comment by Wxh16144** [comment]
考虑一下是否更新一下 BUG_VERSIONS.json


5.22.1: https://github.com/ant-design/ant-design/blob/5.22.1/BUG_VERSIONS.json#L65 

```json
  ">= 5.21.6 <=5.22.1": [
    "https://github.com/ant-design/ant-design/pull/51210#issuecomment-2449051375",
    "https://github.com/ant-design/ant-design/issues/51601"
  ]
```

**Comment by MadCcc** [comment]
> ```json
> ">= 5.21.6 <=5.22.1": [
>     "https://github.com/ant-design/ant-design/pull/51210#issuecomment-2449051375",
>     "https://github.com/ant-design/ant-design/issues/51601"
>   ]
> ```

跟这个 PR 无关吧？

**Review by yoyo837 — APPROVED** [review]

**Pull request merged**